sphinx/addnodes.py

```diff
@@ -115,25 +115,53 @@ def extract_original_messages(self) -> List[str]:
         return messages
 
 
-# domain-specific object descriptions (class, function etc.)
+#############################################################
+# Domain-specific object descriptions (class, function etc.)
+#############################################################
+
+class _desc_classes_injector(nodes.TextElement):
+    """Helper base class for injecting a fixes list of classes.
+
+    Use as the first base class.
+    """
+
+    classes = []  # type: List[str]
```

Comment:
Since v4.0, we can use variable annotations syntax because we drop the py35 support. How about using it?
```suggestion
    classes: List[str] = []
```

Reply:
Indeed, good idea.